ipyparallel/client/client.py

```diff
@@ -1042,6 +1042,11 @@ def _dispatch_iopub(self, msg):
             # ignore status messages if they aren't mine
             return
 
+        # Run any callback functions
+        msg_future = self._futures[msg_id]
+        for callback in msg_future.io_stream_callbacks:
+            callback(msg)
```

Comment:
Probably move this callback to the very end of this function, so it doesn't disrupt any of the other message handling.

Reply:
Makes sense. Done.